iree/compiler/Codegen/Utils/Utils.h

```diff
@@ -133,6 +134,24 @@ LogicalResult getComputeOps(
     FuncOp funcOp, SmallVectorImpl<Operation *> &computeOps,
     SmallVectorImpl<LoopTilingAndDistributionInfo> &tiledLoops);
 
+/// A struct containing function pointers to query processor information.
+/// TODO: This should be replaced by some interface.
```

Comment:
I think its not too hard to add this interface. This interface can live in IREE Codegen dialect for now, and then you can just add external models for all the ops you are interested in. It would make all of this much more clean. Most of the things in this PR are fantastic, so I really want this PR to land, but it seems like just having this interface will save us a lot of trouble.

Reply:
Changed to interfaces and external models.